recipes/cassandra-cpp-driver/all/patches/2.15.3/fix-cmake.patch

```diff
@@ -0,0 +1,352 @@
++  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -pthread")
++endif()
++
++#------------------------
++# Boost
++#------------------------
++
++if(CASS_USE_BOOST_ATOMIC)
++  # Ensure Boost auto linking is disabled (defaults to auto linking on Windows)
++  if(WIN32)
++    add_definitions(-DBOOST_ALL_NO_LIB)
++  endif()
++
++  # Determine if additional Boost definitions are required for driver/executables
++  if(NOT WIN32)
++    # Handle explicit initialization warning in atomic/details/casts
++    add_definitions(-Wno-missing-field-initializers)
++  endif()
++endif()
++
+ file(GLOB SOURCES *.cpp *.hpp)
+ 
+ if(APPLE)
+@@ -38,15 +73,11 @@ endif()
+ 
+ add_subdirectory(third_party/curl)
+ add_subdirectory(third_party/hdr_histogram)
+-add_subdirectory(third_party/http-parser)
+-add_subdirectory(third_party/minizip)
+ add_subdirectory(third_party/sparsehash)
+ 
+ list(APPEND INCLUDE_DIRS 
+   third_party/curl
+   third_party/hdr_histogram
+-  third_party/http-parser
+-  third_party/minizip
+   third_party/mt19937_64
+   third_party/rapidjson/rapidjson
+   third_party/sparsehash/src)
+@@ -126,78 +157,44 @@ configure_file(
+ # Targets
+ #------------------------------
+ 
+-if(CASS_BUILD_SHARED)
+-  add_library(cassandra SHARED 
+-    ${SOURCES}
+-    $<TARGET_OBJECTS:curl_hostcheck>
+-    $<TARGET_OBJECTS:hdr_histogram>
+-    $<TARGET_OBJECTS:http-parser>
+-    $<TARGET_OBJECTS:minizip>)
+-  target_link_libraries(cassandra ${CASS_LIBS})
+-  target_include_directories(cassandra PRIVATE ${INCLUDE_DIRS} ${CASS_INCLUDES})
+-
+-  set_target_properties(cassandra PROPERTIES OUTPUT_NAME cassandra)
+-  set_target_properties(cassandra PROPERTIES VERSION ${PROJECT_VERSION_STRING} SOVERSION ${PROJECT_VERSION_MAJOR})
+-  set_target_properties(cassandra PROPERTIES
+-      COMPILE_PDB_NAME "cassandra"
+-      COMPILE_PDB_OUTPUT_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}")
+-  set_target_properties(cassandra PROPERTIES
+-      ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}
+-      LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}
+-      RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR})
+-  set_target_properties(cassandra PROPERTIES FOLDER "Driver")
+-
+-  target_compile_definitions(cassandra PRIVATE CASS_BUILDING)
+-
+-  if(CASS_USE_BOOST_ATOMIC AND BOOST_LIBRARY_NAME)
+-    add_dependencies(cassandra ${BOOST_LIBRARY_NAME})
+-  endif()
+-  if(LIBUV_LIBRARY_NAME)
+-    add_dependencies(cassandra ${LIBUV_LIBRARY_NAME})
+-  endif()
+-  if(OPENSSL_LIBRARY_NAME)
+-    add_dependencies(cassandra ${OPENSSL_LIBRARY_NAME})
+-  endif()
+-  if(ZLIB_LIBRARY_NAME)
+-    add_dependencies(cassandra ${ZLIB_LIBRARY_NAME})
+-  endif()
++add_library(${PROJECT_LIB_NAME_TARGET}
++  ${SOURCES}
++  $<TARGET_OBJECTS:curl_hostcheck>
++  $<TARGET_OBJECTS:hdr_histogram>)
++
++
++conan_target_link_libraries(${PROJECT_LIB_NAME_TARGET})
++
++if(WIN32)
++ target_link_libraries(${PROJECT_LIB_NAME_TARGET}
++    iphlpapi
++    psapi
++    wsock32
++    crypt32
++    ws2_32
++    userenv
++    version)
+ endif()
```

Comment:
Are these system libraries of cassandra, or from the third-party libraries?

Reply:
I can say for sure that version lib is used at [in](https://github.com/datastax/cpp-driver/blob/3d8c5abc16bc51b49de6e2931848e4c3553480be/src/client_insights.cpp#L126).

As for rest libraries I had a look at bincrafters [recipe ](https://github.com/bincrafters/conan-cassandra-driver/blob/5e11c49acfe4fd23d7c3ac6d0e6ca983e71ec2cd/conanfile.py#L81)

Reply:
See https://github.com/datastax/cpp-driver/blob/3d8c5abc16bc51b49de6e2931848e4c3553480be/cmake/Dependencies.cmake#L7

Reply:
Could you please tell what is wrong with my answer?

Reply:
You answer was completely correct, I just pointed to the upstream list of windows implibs.
Bincrafters recipes are not always 100% accurate and it's good to have some scrutiny.

Reply:
@madebr The system libraries you see in the patch are taken from [the cmake you mentioned](https://github.com/datastax/cpp-driver/blob/3d8c5abc16bc51b49de6e2931848e4c3553480be/cmake/Dependencies.cmake#L7). Since this file is provided by developers of the driver, our team does not know if they are needed for the driver itself or it is that the driver dependencies require them. Should we find out which one of the driver dependencies (or the driver itself) requires which system library? And does that matter in our case?

Reply:
You can if you want to and tell upstream, but it's not worth the time imho.
Linking to many libraries might have speed and ram implications.
But if that is the limiting factor here, then you're having bigger problems.

Reply:
I think it is not not worth the time. I guess we can close "third-party libs" thread. Are you agree?

Reply:
The list is fine. You can *close* this thread